src/cr/cube/cube_slice.py

```diff
@@ -258,4 +259,14 @@ def ndim(self):
 
     @lazyproperty
     def shape(self):
-        return self.as_array().shape
+        warnings.warn('Deprecated. Use `get_shape` instead.', DeprecationWarning)
+        return self.get_shape()
+
+    def get_shape(self, prune=False):
```

Comment:
Please add docstring (PEP 257) describing interface and what a shape is.

Reply:
Done. I've run the `pylint` on the entire file, should be better now...